src/main/java/uk/gov/ida/verifyserviceprovider/configuration/HubMetadataConfiguration.java

```diff
@@ -1,147 +1,70 @@
-     *
-     * This password MUST NOT be used for anything sensitive, since it is open source.
-     */
-    private String DEFAULT_TRUST_STORE_PASSWORD = "bj76LWZ+F5L1Biq4EZB+Ta7MUY4EQMgmZmqAHh";
-
-    private final String trustStorePath;
-    private final String trustStorePassword;
-    private final URI uri;
-    private final Long minRefreshDelay;
-    private final Long maxRefreshDelay;
-    private final String expectedEntityId;
-    private final JerseyClientConfiguration jerseyClientConfiguration;
-    private final String jerseyClientName;
-    private final boolean shouldLoadTrustStoreFromResources;
-
-    public HubMetadataConfiguration(HubEnvironment environment, MetadataConfigurationOverrides overrides) {
-        this.trustStorePath = ofNullable(overrides.getTrustStorePath()).orElseGet(() -> generateTrustStorePath(environment));
-        this.trustStorePassword = ofNullable(overrides.getTrustStorePassword()).orElse(DEFAULT_TRUST_STORE_PASSWORD);
-        this.uri = ofNullable(overrides.getUri()).orElse(environment.getMetadataUri());
-        this.minRefreshDelay = ofNullable(overrides.getMinRefreshDelay()).orElse(60000L);
-        this.maxRefreshDelay = ofNullable(overrides.getMaxRefreshDelay()).orElse(600000L);
-        this.expectedEntityId = ofNullable(overrides.getExpectedEntityId()).orElseGet(() -> generateExpectedEntityId(environment));
-        this.jerseyClientConfiguration = ofNullable(overrides.getJerseyClientConfiguration()).orElse(createClient());
-        this.jerseyClientName = ofNullable(overrides.getJerseyClientName()).orElse(HUB_JERSEY_CLIENT_NAME);
-        this.shouldLoadTrustStoreFromResources = (overrides.getTrustStorePath() == null);
+public class HubMetadataConfiguration extends MetadataConfiguration {
+
+    private HubEnvironment environment;
+    private final TrustStoreConfiguration trustStoreConfiguration;
+
+    @JsonCreator
+    public HubMetadataConfiguration(
+            @JsonProperty("uri") @JsonAlias({"url"}) URI uri,
+            @JsonProperty("minRefreshDelay") Long minRefreshDelay,
+            @JsonProperty("maxRefreshDelay") Long maxRefreshDelay,
+            @JsonProperty("expectedEntityId") String expectedEntityId,
+            @JsonProperty("client") JerseyClientConfiguration client,
+            @JsonProperty("jerseyClientName") String jerseyClientName,
+            @JsonProperty("hubFederationId") String hubFederationId,
+            @JsonProperty("trustStore") TrustStoreConfiguration trustStoreConfiguration) {
+        super(uri, minRefreshDelay, maxRefreshDelay, expectedEntityId, client,
+                ofNullable(jerseyClientName).orElse(HUB_JERSEY_CLIENT_NAME), hubFederationId);
+        this.trustStoreConfiguration = trustStoreConfiguration;
     }
 
-    @NotNull
-    @Valid
-    @Override
-    public String getTrustStorePath() {
-        return trustStorePath;
+    public HubMetadataConfiguration() {
+        this(null, null, null, null, null, null, null, null);
```

Comment:
This is just to make the `ofNullable(hubMetadataConfiguration).orElse(new HubMetadataConfiguration())` read a bit nicer but it is pretty ugly overall. Maybe a factory method somewhere would be nicer?